h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -968,17 +990,24 @@ private FVecParseWriter streamParse(final InputStream is, final ParseSetup local
           break;
         }
         if ((_setup.getParseType().name().toLowerCase().equals("svmlight") ||
-                (_setup.getParseType().name().toLowerCase().equals("avro") ))
-                && ((_setup.getSkippedColumns() != null) && (_setup.getSkippedColumns().length >0)))
+                (_setup.getParseType().name().toLowerCase().equals("avro")))
+                && ((_setup.getSkippedColumns() != null) && (_setup.getSkippedColumns().length > 0)))
           throw new H2OIllegalArgumentException("Parser: skipped_columns are not supported for " +
                   "SVMlight or Avro parsers.");
 
-        if (_setup.getSkippedColumns() !=null &&
-                ((_setup.get_parse_columns_indices()==null) || (_setup.get_parse_columns_indices().length==0)))
+        if (_setup.getSkippedColumns() != null &&
+                ((_setup.get_parse_columns_indices() == null) || (_setup.get_parse_columns_indices().length == 0)))
           throw new H2OIllegalArgumentException("Parser:  all columns in the file are skipped and no H2OFrame" +
                   " can be returned."); // Need this to send error message to R
-
         p.parseChunk(in.cidx(), din, dout);
+        for (int partitionIndex = 0; partitionIndex < _partitionedBy.length; partitionIndex++) {
```

Comment:
This piece of code will not work, as the lines are already there in the `dout` writer. The partitioned  columns must be written either:

1. During the process of parsing itself - which means inserting partition columns into each parser and writing support for it (much of it could be abstracted away to the Abstract Parser class).
2. Do it after the Parsing (Distributed/Sequential). This would bypass some type inferences and checks and data into the missing columns would have to be filled manually.

There is also the question of how to treat column indices - lie to the parsers there are actually less columns and add the partitioned columns afterwards, or not lie to it make sure the missing columns are skipped corretly by the parser - making sure those columns are at the end as they're now should do the trick.